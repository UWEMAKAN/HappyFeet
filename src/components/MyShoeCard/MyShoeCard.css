.MyShoeCard {
  display: flex;
  height: 20rem;
  width: 100%;
  padding: 2rem 10rem;
  border: 1px solid rgba(0,0,0,.2);
}

.MyShoeCard:not(:last-child) {
  margin-bottom: 5rem;
}

.MyShoeCard img {
  width: 16rem;
}

.MyShoeCard .Description {
  align-self: center;
  display: flex;
  margin-left: 18rem;
}

.MyShoeCard .Description div:first-child {
  display: flex;
  flex-direction: column;
  width: 37rem;
}

.MyShoeCard .Description span:first-child {
  font-size: 2rem;
}

.MyShoeCard .Description span:not(:first-child) {
  color: #887d7d;
}

.MyShoeCard .Description span:not(:last-child) {
  margin-bottom: 1rem;
}

.MyShoeCard .Price {
  width: 15rem;
  font-size: 2rem;
}
