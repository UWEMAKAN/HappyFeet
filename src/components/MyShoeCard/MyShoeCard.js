import React from 'react';
import propTypes from 'prop-types';
import Image from '../Common/Image/Image';
import './MyShoeCard.css';

const MyShoeCard = ({
  image, title, color, seller, price
}) => (
  <div className="MyShoeCard">
    <Image image={image} title={title} />
    <div className="Description">
      <div>
        <span>
          <strong>{title}</strong>
        </span>
        <span>
          Color:
          {' '}
          {color}
        </span>
        <span>
          Seller:
          {' '}
          {seller}
        </span>
      </div>
      <div className="Price">
        <strong>
          Rs.
          {' '}
          {price}
        </strong>
      </div>
    </div>
  </div>
);

MyShoeCard.propTypes = {
  image: propTypes.string.isRequired,
  title: propTypes.string.isRequired,
  color: propTypes.string.isRequired,
  seller: propTypes.string.isRequired,
  price: propTypes.number.isRequired
};

export default MyShoeCard;
