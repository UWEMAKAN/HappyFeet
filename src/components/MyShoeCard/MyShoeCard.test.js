import React from 'react';
import { shallow } from 'enzyme';
import MyShoeCard from './MyShoeCard';

const renderMyShoeCard = (args) => {
  const defaultProps = {
    image: '',
    title: '',
    color: '',
    seller: '',
    price: ''
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<MyShoeCard {...props} />);
};

describe('MyShoeCard Component', () => {
  test('should render MyShoeCard Component', () => {
    const image = 'http://localhost:3001/assets/images/image.jpg';
    const title = 'Nike White Snickers for Men and Women';
    const color = 'black';
    const seller = 'RetailNet';
    const price = 3500;
    const wrapper = renderMyShoeCard({
      image, title, color, seller, price
    });

    expect.assertions(2);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getElements()[0].props.children.length).toBe(2);
  });
});
