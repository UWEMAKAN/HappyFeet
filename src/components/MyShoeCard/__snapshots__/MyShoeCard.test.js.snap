// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyShoeCard Component should render MyShoeCard Component 1`] = `ShallowWrapper {}`;
